### ISSUE TITLE
why there is a re-implementation of gradient checkpointing

### ISSUE THREAD

**Issue by zen-d** [MCG-NJU/SparseBEV#52]
HI @afterthat97, thanks for your awesome work. I noticed there is a `checkpoint.py` file that seems to re-implement `torch.utils.checkpoint`. What is the motivation for that?

Furthermore, could you please provide your reference pytorch implementation version so that I could get a quick diff (since pytorch is rapidly developing, the specific implementation may vary from version to version)? Thanks.

**Comment by afterthat97** [comment]
This implementation is exactly the same as the official code of torch 2.0. So, if you are using torch2.0, you can directly use torch.utils.checkpoint.

However, for torch1.12 users, the official implementation does not support use_reentrant=True, so you may need this copy.

**Comment by afterthat97** [comment]
Yes you are correct.

Our checkpoint.py is directly copied from https://pytorch.org/docs/2.0/_modules/torch/utils/checkpoint.html#checkpoint with no modifications
